strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/StrongboxBuiltinAuthenticationSupplier.java

```diff
@@ -0,0 +1,95 @@
+package org.carlspring.strongbox.authentication.api.impl;
+
+import org.carlspring.strongbox.authentication.api.AuthenticationSupplier;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+
+import com.google.common.base.Throwables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.crypto.codec.Base64;
+import org.springframework.util.Assert;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class StrongboxBuiltinAuthenticationSupplier
```

Comment:
I think DefaultAuthenticationSupplier is best choice of the three you proposed.
 
This is the class that uses Basic Authentication check for already existing strongbox authentication mechanism. See https://youtrack.carlspring.org/issue/SB-625 for reference and your comment:

> Currently, the users and their permissions are loaded from an XML file. This is the built-in authentication. 

and this class fits the existing implementation where the user was taken from Basic HTTP header.

That's why I called it StrongboxBuiltinAuthenticationSupplier

I'll change it to DefaultAuthenticationSupplier, as you suggested